core/trino-main/src/main/java/io/trino/operator/GroupedTopNBuilder.java

```diff
@@ -27,4 +27,10 @@
     Iterator<Page> buildResult();
 
     long getEstimatedSizeInBytes();
+
+    boolean isFull();
+
+    boolean isFlushable();
+
+    int getCapacity();
```

Comment:
I would recommend the following simplification:
1. Just add a single isFull() API
2. Always update the full check regardless of whether it is flushable or not. The comparison is cheap.
3. Get rid of getCapacity() from TopNRankingOperator (is this even used)?
4. Use DataSize instead of Optional<DataSize>, and set it to be DataSize of Long.MAX_VALUE bytes if it is a partial aggregation.

Reply:
> For 3, this is used for testing.
I mean are there tests that use this? I might have overlooked them, but didn't see any after a quick scan
> For 4, do you mean Long.MAX_VALUE if it is a final(not partial) aggregation? I would prefer using long if Optional needs to be replaced. So, DataSize doesn't need to be converted to a long type value every time it is used.
Yes, I just mean to use the MAX_VALUE when we don't need the behavior to do anything (e.g. in final agg). Long is fine too, or you can have the constructor take a DataSize to ensure type safety, and store the long bytes internally.

Reply:
Done.